### PR TITLE
Adding GREG vagrant sample and updating hiera files

### PR DESCRIPTION
**Pull request by Thanu**
This p/r is to update the GREG default platform hiera yaml files and add vagrant sample.

### CODE REVIEW

File: hieradata/dev/wso2/wso2greg/5.1.0/default/default.yaml [hunk 1]
```diff
@@ -16,34 +16,40 @@ classes:
...
 
 wso2::is_datasource: wso2_carbon_db
 
 wso2::master_datasources:
+# Comment the below h2 datasources for clustered setup
   wso2am_db:
     name: WSO2AM_DB
     description: The datasource used for API Manager database
-    driver_class_name: org.h2.Driver
+    driver_class_name: "%{hiera('wso2::datasources::mysql::driver_class_name')}"
```
Comment (ravihansa3000) [inline hieradata/dev/wso2/wso2greg/5.1.0/default/default.yaml]: shouldn't this be h2 driver?

File: hieradata/dev/wso2/wso2greg/4.6.0/default/default.yaml [hunk 2]
```diff
@@ -16,34 +16,39 @@ classes:
...
 wso2::pack_extracted_dir: "%{::product_name}-%{::product_version}"
+wso2::datasources::common::bps_default_auto_commit: 'true'
 
 wso2::service_name: wso2greg
 wso2::hostname: greg.dev.wso2.org
 wso2::mgt_hostname: greg.dev.wso2.org
 
+# Uncomment MySQL connector driver for a clustered setup
 #wso2::file_list:
 #  - repository/components/lib/mysql-connector-java-5.1.36-bin.jar
```
Comment (ravihansa3000) [inline hieradata/dev/wso2/wso2greg/4.6.0/default/default.yaml]: connector jar name should be another sub Hiera lookup

Comment (Thanu) [inline hieradata/dev/wso2/wso2greg/4.6.0/default/default.yaml]: Fixed in commit 6c3a01f42a1ef9dfad4be0d5ebdee5c46e7eea26.

File: hieradata/dev/wso2/wso2greg/5.1.0/default/default.yaml (hunk 1 above)
Comment (Thanu) [inline hieradata/dev/wso2/wso2greg/5.1.0/default/default.yaml]: Fixed in commit 6c3a01f42a1ef9dfad4be0d5ebdee5c46e7eea26.

**Comment by chamilad** [comment]
Merged. Thanks!

**Pull request merged**